nestkernel/conn_builder.h

```diff
@@ -490,6 +506,8 @@ class SPBuilder : public ConnBuilder
   void sp_connect( const std::vector< size_t >& sources, const std::vector< size_t >& targets );
 
 protected:
+  std::string name_;
```

Comment:
What is this `name_`?

Reply:
This is the name of the `ConnBuilder`. I admit that this is a bit of a hack, but I didn't know where else to store this variable.

This name is needed to be able to return the same names when retrieving the kernel attribute `structural_plasticity_synapses` as used when setting them. Previously, a user set them using a name, e.g.
```
nest.structural_plasticity_synapses = { "synapse_ex": { ...},  "synapse_in": { ...}  }
```
but got back a numbered, e.g.
```
nest.structural_plasticity_synapses = { "syn1": { ...},  "syn2": { ...}  }
```
which was a) confusing and b) error prone due to the likely loss of ordering.

Now, the name is kept in the `ConnBuilder` object, so it can be used for the retrieval of parameters.

Reply:
Added in 3aa649a5a.